Documentation/Evolution/RegexSyntax.md

```diff
@@ -0,0 +1,245 @@
+
+#### Anchors
+
+```
+Anchor -> '^' | '$' | '\b'
+```
+
+Anchors match against a certain position in the input rather than on a particular character of the input.
+
+#### Unicode scalars
+
+
+
+#### Builtin character classes
+
+
+
+#### Custom character classes
+
+```
+CustomCharClass -> Start Set (SetOp Set)* ']'
+Start           -> '[' '^'?
+Set             -> Member+
+Member          -> CustomCharClass | !']' !SetOp (Range | Atom)
+Range           -> Atom `-` Atom
+```
+
+Custom characters classes introduce their own language, in which most regular expression metacharacters become literal
+
+
+#### Character properties
+
+### PCRE-specific syntax
+
+#### Callouts
+
+### Oniguruma-specific syntax
+
+#### Custom reference syntax
+
+#### Callout syntax
+
+#### Absent functions
+
+### ICU-specific syntax
+
+
+
+### .NET-specific syntax
+
+#### Balancing groups
+
+```
+GroupNameBody -> Identifier | Identifier? '-' Identifier
+```
+
+.NET supports the ability for a group to reference a prior group, causing the prior group to be deleted, and any intermediate matched input to become the capture of the current group.
+
+#### Character class subtraction with `-`
+
+
+
+## Syntactic differences between engines
+
+### Conflicting differences
+
+#### Character class set operations
+
+In a custom character class, some engines allow for binary set operations that take two character class inputs, and produce a new character class output. However which set operations are supported and the spellings used differ by engine.
+
+| PCRE | ICU | UTS#18 | Oniguruma | .NET | Java |
+|------|-----|--------|-----------|------|------|
+| ❌ | Intersection `&&`, Subtraction `--` | Intersection & Subtraction | Intersection `&&` | Subtraction via `-` | Intersection  `&&` |
+
+[UTS#18][uts18] requires intersection and subtraction, and uses the operation spellings `&&` and `--` in its examples, though it doesn't mandate a particular spelling. In particular, conforming implementations could spell the subtraction `[[x]--[y]]` as `[[x]&&[^y]]`. UTS#18 also suggests a symmetric difference operator `~~`, and uses an explicit `||` operator in examples, though doesn't require either operations.
+
+These differences are conflicting, as engines that don't support a particular operator treat them as literal, e.g `[x&&y]` in PCRE is the character class of `["x", "&", "y"]` rather than an intersection.
+
+We intend to support the operators `&&`, `--`, `-`, and `~~`. This means that any regex literal containing these sequences in a custom character class while being written for an engine not supporting that operation will have a different semantic meaning in our engine. However this ought not to be a common occurrence, as specifying a character multiple times in a custom character class is redundant. However, we intend on providing a strict compatibility mode that may be used to emulate behavior of a particular engine (**TODO: all engines, or just PCRE?**).
+
```

Comment:
I feel like the double-characters are no-brainers for us and should be their own paragraph. I think the single character `-` is a little more arguable, so I'd put it in its own paragraph and be up front with that fact.